test/optim/test_initializers.py

```diff
@@ -275,14 +275,14 @@ def test_gen_batch_initial_conditions_constraints(self):
             bounds = torch.tensor([[0, 0], [1, 1]], device=self.device, dtype=dtype)
             inequality_constraints = [
                 (
-                    torch.tensor([1], device=self.device, dtype=dtype),
+                    torch.tensor([1], device=self.device, dtype=torch.int64),
```

Comment:
Hmm, so non-int dtypes used to be acceptable in `gen_batch_initial_conditions` and `get_polytope_samples`, but now a non-int will cause a failure in `normalize_linear_constraints`. I'd prefer not to break backwards compatibility, but this should be fine for now since it isn't breaking any tests. I'll add this to the issue.

Reply:
I think it's fine to break backward-compatibility here; it's kind of dicey to use floats to index tensor positions - more likely than not code that was doing that didn't actually mean to do that (that was the case with a type bug in our test suite as well).